File: backend/src/main/java/z9/hobby/domain/authentication/service/AuthenticationService.java
package z9.hobby.domain.authentication.service;

import z9.hobby.domain.authentication.dto.AuthenticationRequest;
import z9.hobby.domain.authentication.dto.AuthenticationResponse;

public interface AuthenticationService {

    AuthenticationResponse.UserToken login(AuthenticationRequest.Login dto);

    AuthenticationResponse.UserToken oauthLogin(String provider, String authCode);

    void signup(AuthenticationRequest.Signup signupDto);

    void logout(String userId);

    void resign(String userId);
}

File: backend/src/main/java/z9/hobby/domain/authentication/service/AuthenticationService.kt
package z9.hobby.domain.authentication.service

import z9.hobby.domain.authentication.dto.AuthenticationRequest
import z9.hobby.domain.authentication.dto.AuthenticationResponse

interface AuthenticationService {

    fun login(dto: AuthenticationRequest.Login): AuthenticationResponse.UserToken

    fun oauthLogin(provider: String, authCode: String): AuthenticationResponse.UserToken

    fun signup(signupDto: AuthenticationRequest.Signup)

    fun logout(userId: String)

    fun resign(userId: String)
}
